src/app/app.component.ts

```diff
@@ -1,13 +1,21 @@
-import { mockedCoursesList } from './shared/mocks/mocks';
-import { Component } from '@angular/core';
+import { ButtonLabelService } from "./services/button-label.service";
+import { mockedCoursesList } from "./shared/mocks/mocks";
+import { Component } from "@angular/core";
 
 @Component({
-  selector: 'app-root',
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  selector: "app-root",
+  templateUrl: "./app.component.html",
+  styleUrls: ["./app.component.scss"],
 })
 export class AppComponent {
-  title = 'courses-app';
-
+  title = "courses-app";
   mockedCoursesList = mockedCoursesList;
+
+  loginButtonLabel!: string;
+  newCourseButtonLabel!: string;
+
+  constructor(private buttonLabelService: ButtonLabelService) {
```

Comment:
Ok, thanks for introducing new logic for button text handling. I just have one more simplification request. By marking the ButtonLabelService with public (or protected) visibility, you will be able to read the service's contents from Compontent template's. This will eliminate the need to add `loginButtonLabel` and `newCourseButtonLabel` variables.
Use the service like this in the component template: `<app-button [buttonText]="buttonLabelService.getLabel("LOGIN")">`